doc/contribute/doc-guidelines.rst

```diff
@@ -0,0 +1,308 @@
+      * so it doesn't
+      * use up so much
+      * space on
+      * the page
+
+This would be rendered as:
+
+.. hlist::
+   :columns: 3
+
+   * A list of
+   * short items
+   * that should be
+   * displayed
+   * horizontally
+   * so it doesn't
+   * use up so much
+   * space on
+   * the page
+
+Note the optional ``:columns:`` parameter (default is two columns), and
+all the list items are indented by three spaces.
+
+File names and Commands
+***********************
+
+Sphinx extends ReST by supporting additional inline markup elements (called
+"roles") used to tag text with special
+meanings and allow style output formatting. (You can refer to the `Sphinx Inline Markup`_
+documentation for the full list).
+
+For example, there are roles for marking :file:`filenames`
+(``:file:\`name\```) and command names such as :command:`make`
+(``:command:\`make\```).  You can also use the \`\`inline code\`\`
+markup (double backticks) to indicate filenames.
+
+.. _internal-linking:
+
+Internal Cross-Reference Linking
+********************************
+
+ReST links are only supported within the current file using the
+notation ``refer to the `internal-linking`_ page`` (which renders as,
+refer to the `internal-linking`_ page).  Note the use of a trailing
+underscore to indicate an outbound link. In this example, the label was
+added immediately before a heading, so the text that's displayed is the
+heading text itself.
+
+With Sphinx however, we can create
+link-references to any tagged text within the Zephyr Project documentation.
+
+Target locations within documents are defined with a label directive:
+
+   .. code-block:: rst
+
+      .. _my label name:
+
+Note the leading underscore (``\_``) indicating an inbound link.
+The content immediately following
+this label is the target for a ``\:ref:\`my label name\```
+reference from anywhere within the Zephyr documentation.
+The label should be added immediataly before a heading so there's a
+natural phrase to show when referencing this label (e.g., the heading
+text).
+
+This is the same directive used to
+define a label that's a reference to an URL::
+
+   .. _Zephyr Wikipedia Page:
+      https://en.wikipedia.org/wiki/Zephyr_(operating_system)
+
+
+To enable easy cross-page linking within the site, each file should have
+a reference label before its title so it may
+be referenced from another file. These reference labels must be unique
+across the whole site, so generic names such as "samples" should be
+avoided.  For example the top of this .rst file is:
+
+
+.. code-block:: rst
+
+   .. _doc_guidelines:
+
+   Documentation Guidelines for the Zephyr Project
+   ###############################################
+
+
+Other .rst documents can link to this document using the ``:ref:`doc_guidelines``` tag and
+it will show up as :ref:`doc_guidelines`.  This type of internal cross reference works across
+multiple files, and the linking text is obtained from the document source so if the title changes,
```

Comment:
Here, it says "linking text", but the next line says "link text". I think "link text" is the common usage, but either way, it might be more clear to be consistent here.